src/hooks/useUserLocation.ts

```diff
@@ -0,0 +1,50 @@
+import * as LocationLib from "expo-location";
+import isEqual from "lodash/isEqual";
+import { useState, useEffect, useRef } from "react";
+
+const requestLocationPermission = async () => {
+  let { status } = await LocationLib.requestForegroundPermissionsAsync();
+  if (status !== "granted") {
+    throw new Error("Permission to access location was denied");
+  }
+};
+
+const useUserLocation = () => {
+  const intervalRef = useRef<NodeJS.Timeout | null>(null);
+  const [location, setLocation] = useState<LocationLib.LocationObject | null>(
+    null
+  );
+  const [errorMsg, setErrorMsg] = useState<string | null>(null);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        await requestLocationPermission();
+        intervalRef.current = setInterval(async () => {
+          const location = await LocationLib.getCurrentPositionAsync({});
+
+          setLocation((prev) => {
+            if (isEqual(prev?.coords, location.coords)) {
```

Comment:
Try to think about code readability: you have 5 levels of nested functions here 🙈

Reply:
overall it's always better to put named function as callback 👍